src/lib/version-util.ts

```diff
@@ -523,22 +519,22 @@ export function upgradeDependencyDeclaration(
       (isGreaterThan ? '>=' : operator) + version
 }
 
-/** Reverts a valid semver version to a pseudo version that is missing its minor and patch components. NOOP If the original version was a valid semver version. */
-const revertMissingMinorAndPatch = curry((current: string, latest: string) =>
-  isMissingMinorAndPatch(current) ? latest.slice(0, latest.length - '.0.0'.length) : latest,
-)
+/** Reverts a valid semver version to a pseudo version. NOOP If the original version was a valid semver version. */
+const revertPseudoVersion = (current: string, latest: string) => {
+  /** Reverts a valid semver version to a pseudo version with a leading 'v'. NOOP If the original version was a valid semver version. */
+  const leadingV = v(current)
+  let result = leadingV ? leadingV + latest : latest
 
-/** Reverts a valid semver version to a pseudo version that is missing its patch components. NOOP If the original version was a valid semver version. */
-const revertMissingPatch = curry((current: string, latest: string) =>
-  isMissingPatch(current) ? latest.slice(0, latest.length - '.0'.length) : latest,
-)
+  /** Reverts a valid semver version to a pseudo version that is missing its minor and patch components. NOOP If the original version was a valid semver version. */
+  const missingMinorAndPatch = isMissingMinorAndPatch(current)
+  result = missingMinorAndPatch ? result.slice(0, result.length - '.0.0'.length) : result
 
-/** Reverts a valid semver version to a pseudo version with a leading 'v'. NOOP If the original version was a valid semver version. */
-const revertLeadingV = curry((current: string, latest: string) => (v(current) ? v(current) + latest : latest))
+  /** Reverts a valid semver version to a pseudo version that is missing its patch components. NOOP If the original version was a valid semver version. */
+  const missingPatch = isMissingPatch(current)
+  result = missingPatch ? result.slice(0, result.length - '.0'.length) : result
 
-/** Reverts a valid semver version to a pseudo version. NOOP If the original version was a valid semver version. */
-const revertPseudoVersion = (current: string, latest: string) =>
-  flow(revertLeadingV(current), revertMissingMinorAndPatch(current), revertMissingPatch(current))(latest)
+  return result
+}
```

Comment:
I'll accept this change, though a slightly preferable approach would have been to implement a simple flow function and preserve the existing structure of the code. (The currying was clearly unnecessary though.)